lib/manager/terraform/lockfile/hash.ts

```diff
@@ -3,168 +3,125 @@ import extract from 'extract-zip';
+    return result;
   }
-  const lookupName = repositoryRegexResult.groups.lookupName;
-  const backendLookUpName = `terraform-provider-${lookupName}`;
-
-  const cacheKey = `${TerraformProviderDatasource.defaultRegistryUrls[1]}/${repository}/${lookupName}-${version}`;
-  const cachedRelease = await packageCache.get<string[]>(
-    'terraform-provider-release',
-    cacheKey
-  );
-  // istanbul ignore if
-  if (cachedRelease) {
-    return cachedRelease;
+
+  static async hashOfZipContent(
+    zipFilePath: string,
+    extractPath: string
+  ): Promise<string> {
+    await extract(zipFilePath, { dir: extractPath });
+    const files = await fs.readdir(extractPath);
+    // the h1 hashing algorithms requires that the files are sorted by filename
+    const sortedFiles = files.sort((a, b) => a.localeCompare(b));
+    const filesWithPath = sortedFiles.map((file) => `${extractPath}/${file}`);
+
+    const result = await TerraformProviderHash.hashFiles(filesWithPath);
+
+    // delete extracted files
+    await fs.rm(extractPath, { recursive: true });
+
+    return result;
   }
-  let versionReleaseBackend: VersionDetailResponse;
-  try {
-    versionReleaseBackend = await getReleaseBackendIndex(
-      backendLookUpName,
-      version
-    );
-  } catch (err) {
-    logger.debug(
-      { err, backendLookUpName, version },
-      `Failed to retrieve builds for ${backendLookUpName} ${version}`
+
+  @cache({
+    namespace: `datasource-${TerraformProviderDatasource.id}-build-hashes`,
+    key: (build: TerraformBuild) => build.url,
+    ttlMinutes: TerraformProviderHash.hashCacheTTL,
+  })
+  static async calculateSingleHash(
+    build: TerraformBuild,
+    cacheDir: string
+  ): Promise<string> {
+    const downloadFileName = join(cacheDir, build.filename);
+    const extractPath = join(cacheDir, 'extract', build.filename);
+    logger.trace(
+      `Downloading archive and generating hash for ${build.name}-${build.version}...`
     );
-    return null;
+    const readStream = TerraformProviderHash.http.stream(build.url);
+    const writeStream = fs.createWriteStream(downloadFileName);
+
+    let hash;
+    try {
+      await fs.pipeline(readStream, writeStream);
+
+      hash = await this.hashOfZipContent(downloadFileName, extractPath);
+      logger.trace(
+        { hash },
+        `Generated hash for ${build.name}-${build.version}`
+      );
+    } finally {
+      // delete zip file
+      await fs.unlink(downloadFileName);
+    }
+    return hash;
   }
 
-  const builds = versionReleaseBackend.builds;
-  const hashes = await calculateHashes(builds);
+  static async calculateHashes(builds: TerraformBuild[]): Promise<string[]> {
+    const cacheDir = await ensureCacheDir('./others/terraform');
 
-  // if a hash could not be produced skip caching and return null
-  if (hashes.some((value) => value == null)) {
-    return null;
+    // for each build download ZIP, extract content and generate hash for all containing files
+    return pMap(
```

Comment:
```suggestion
    return await pMap(
```

We should make sure computation is done here.